service/src/main/java/bio/terra/cbas/dependencies/wes/CromwellService.java

```diff
@@ -33,4 +41,16 @@ public RunStatus runStatus(String runId) throws ApiException {
   public Object getOutputs(String id) throws ApiException {
     return cromwellClient.wesAPI().getRunLog(id).getOutputs();
   }
+
+  public String getRunErrors(Run run) throws ApiException {
+    ApiClient client = new ApiClient();
+    client.setBasePath(this.cromwellConfig.baseUri());
+    WorkflowsApi cromwellApi = new WorkflowsApi(client);
+
+    WorkflowMetadataResponse meta =
+        cromwellApi.metadata(
+            "v1", run.engineId(), Collections.singletonList("failure"), null, null);
+
+    return meta.getFailures().get(0).getCausedBy().get(0).getMessage();
```

Comment:
This is risky. We should protect against some perhaps unfortunate (but perfectly possible) responses from Cromwell:
- What if the response failures is an empty array?
- What if the failures object's `causedBy` array is empty?

Reply:
(and of course, add tests for both cases. Since what we're testing is "can we handle a given `WorkflowMetadataResponse` value neatly", it might be nice to create a `metadataResponseToErrorString` method, in which case it should be pretty easy to set up unit tests which generate various `WorkflowMetadataResponse` objects and check that we can get a nice looking error message string out of them.)

Reply:
A quick question on this, can you think of an example of when this may happen? I'm curious as to a situation when something fails, but there is no failure message to accompany it.